datadog/dogstatsd/base.py

```diff
@@ -552,9 +552,11 @@ def _should_flush(self, length_to_be_added):
         return False
 
     def _flush_buffer(self):
-        self._send_to_server("\n".join(self.buffer))
-        self._current_buffer_total_size = 0
-        self.buffer = []
+        with Lock():
```

Comment:
Just in case, that definitely will not work. I'll leave you as an exercise to guess why? 😉

Reply:
what, every thread deserves its own lock! ;)